httpMiddleware.go

```diff
@@ -0,0 +1,199 @@
+		statusCode := statusWriter.statusCode
+
+		logMessage := fmt.Sprintf("[%d] %s %s", statusCode, r.Method, r.URL.String())
+		logMap := map[string]interface{}{
+			"responseTime": fmt.Sprintf("%d", duration),
+		}
+
+		logger(logMessage, logMap)
+	})
+}
+
+func inboundMetricsMiddleware(promRegisterer prometheus.Registerer, h http.Handler) http.Handler {
+
+	httpRequests := prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "inbound_http_status_code_total",
+			Help: "Counter of HTTP status codes of inbound HTTP requests",
+		},
+		[]string{"status_code", "method"},
+	)
+	promRegisterer.MustRegister(httpRequests)
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		statusWriter := &statusCodeResponseWriter{w, http.StatusOK}
+		//start := time.Now()
+		h.ServeHTTP(statusWriter, r)
+
+		// request duration in miliseconds
+		//duration := time.Since(start).Nanoseconds() / 1e6
+		statusCode := statusWriter.statusCode
+
+		httpRequests.WithLabelValues(strconv.Itoa(statusCode), r.Method).Inc()
+
+	})
+}
+
+type roundTripperFunc func(*http.Request) (*http.Response, error)
+type ClientMiddleware func(http.RoundTripper) http.RoundTripper
+
+func (fn roundTripperFunc) RoundTrip(r *http.Request) (*http.Response, error) {
+	return fn(r)
+}
+
+func clientMetricsMiddleware(promRegisterer prometheus.Registerer, requestDestination string) func(http.RoundTripper) http.RoundTripper {
+
+	metricHTTPRequestsStatus := prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name:        "outbound_http_status_code_total",
+			Help:        "Counter of HTTP status codes of outbound HTTP requests",
+			ConstLabels: prometheus.Labels{"destination": requestDestination},
+		},
+		[]string{"status_code"},
+	)
+
+	promRegisterer.MustRegister(metricHTTPRequestsStatus)
+
+	return func(next http.RoundTripper) http.RoundTripper {
+		return roundTripperFunc(func(r *http.Request) (*http.Response, error) {
+
+			res, err := next.RoundTrip(r)
+			if err != nil {
+				return res, err
+			}
+			if res == nil {
+				return res, err
+			}
+
+			metricHTTPRequestsStatus.WithLabelValues(strconv.Itoa(res.StatusCode)).Inc()
+
+			return res, err
+		})
+	}
+}
+
+func githubMetricsMiddleware(promRegisterer prometheus.Registerer) githubapp.ClientMiddleware {
+
+	// Headers from https://developer.github.com/v3/#rate-limiting
+	metricGithubRatelimitLimit := prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Name: "github_ratelimit_limit_info",
+			Help: "Gauge of Github X-RateLimit-Limit header. The maximum number of requests you're permitted to make per hour",
+		},
+	)
+	metricGithubRatelimitRemaining := prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Name: "github_ratelimit_remaining_info",
+			Help: "Gauge of Github X-RateLimit-Remaining header. The number of requests remaining in the current rate limit window",
+		},
+	)
+	metricGithubRatelimitReset := prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Name: "github_ratelimit_reset_UNIX_time",
```

Comment:
This naming is a bit unexpected. Can we make it all lower case? And maybe still with `_info` as suffix?

Reply:
Maybe unexpected, but I thought it was a good idea to include the unit used as per [good practice alleged by prometheus](https://prometheus.io/docs/practices/naming/#metric-names). You think it should be `github_ratelimit_reset_info`? https://developer.github.com/v3/#rate-limiting

Reply:
No no. That is great to have there. Just keep it lower cased :) `github_ratelimit_reset_unix_time_info`

Reply:
Alright 🙂 https://github.com/lunarway/release-manager-bot/pull/21/commits/4d58712f578329be961540312db53fec79bb6e9c